openpype/pipeline/colorspace.py

```diff
@@ -601,3 +606,133 @@ def _get_imageio_settings(project_settings, host_name):
     imageio_host = project_settings.get(host_name, {}).get("imageio", {})
 
     return imageio_global, imageio_host
+
+
+def get_colorspace_settings_from_publish_context(context_data):
+    """Returns solved settings for the host context.
+
+    Args:
+        context_data (publish.Context.data): publishing context data
+
+    Returns:
+        tuple | bool: config, file rules or None
+    """
+    if "imageioSettings" in context_data:
+        return context_data["imageioSettings"]
+
+    project_name = context_data["projectName"]
+    host_name = context_data["hostName"]
+    anatomy_data = context_data["anatomyData"]
+    project_settings_ = context_data["project_settings"]
+
+    config_data = get_imageio_config(
+        project_name, host_name,
+        project_settings=project_settings_,
+        anatomy_data=anatomy_data
+    )
+
+    # in case host color management is not enabled
+    if not config_data:
+        return None
+
+    file_rules = get_imageio_file_rules(
+        project_name, host_name,
+        project_settings=project_settings_
+    )
```

Comment:
I would suggest 2 changes:
1. Always return tuple of 2 items so you can do everywhere
    `config_data, file_rules = get_colorspace_settings_from_publish_context(...)`
2. Cache invalid state too, so it's not recalculated all the time

```suggestion
    file_rules = None
    if config_data:
        file_rules = get_imageio_file_rules(
            project_name, host_name,
            project_settings=project_settings_
        )
```